### PR TITLE
chore(Java): add fold-linked-list

### PR DESCRIPTION
**Pull request by kartickg33**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/Java/linked-lists/fold-linked-list.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+public static class Node {
+    int data;
+    Node next;
+}
```
Comment (aayushjain7) [inline algorithms/Java/linked-lists/fold-linked-list.java:4]: Node constructer is missing.

Comment (aayushjain7) [reply algorithms/Java/linked-lists/fold-linked-list.java:4]: Also, give a description of the question at top of the file before starting the program.

File: algorithms/Java/linked-lists/fold-linked-list.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        ll.addLast(i);
+    }
+    // LinkedList --> [1,2,3,4,5,6] for  n = 6
+
+    System.out.print("Linked List before folding --> ");
+    ll.display();
+    ll.fold();
+    System.out.print("Linked List after folding --> ");
+    ll.display();
+}
```
Comment (aayushjain7) [inline algorithms/Java/linked-lists/fold-linked-list.java:83]: Write the output of the program also.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]
Also, update the README.md file of the Java folder,

File: algorithms/Java/linked-lists/fold-linked-list.java (hunk 2 above)
Comment (kartickg33) [reply algorithms/Java/linked-lists/fold-linked-list.java]: Updated as required.

File: algorithms/Java/linked-lists/fold-linked-list.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+//    Java implementation for folding a linked list
+public static class Node {
+    int data;
+    Node next;
+    public Node(int data){
+        this.data = data;
+    }
+}
```
Comment (aayushjain7) [inline algorithms/Java/linked-lists/fold-linked-list.java:8]: set for next also
`this.next = null`

File: algorithms/Java/linked-lists/fold-linked-list.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
+//    Java implementation for folding a linked list
+public static class Node {
+    int data;
+    Node next;
+    public Node(int data){
+        this.data = data;
```
Comment (aayushjain7) [inline algorithms/Java/linked-lists/fold-linked-list.java:6]: set for next also,
`this.next = null;`

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

File: algorithms/Java/linked-lists/fold-linked-list.java (hunk 4 above)
Comment (kartickg33) [reply algorithms/Java/linked-lists/fold-linked-list.java:6]: updated

**Review by aayushjain7 — APPROVED** [review]
Nice work! 👍🏼

**Comment by kartickg33** [comment]
Please review my PR

**Pull request merged**